package/butterfly/lib/dev_server.dart

```diff
@@ -123,29 +120,36 @@ class ButterflyDevServer {
     request.response.write(JSON.encode(result));
   }
 
-  Future<Null> _serveDevServerClientFile(HttpRequest request) async {
+  /// Mimics pub serve's behavior of service package-relative static files.
+  Future<Null> _servePackageAsset(HttpRequest request) async {
     final packagesFile = new File('.packages');
 
     if (!await packagesFile.exists()) {
       throw new StateError('.packages file not found. Have you run pub get?');
     }
 
+    final pathFragments = pathlib.split(request.uri.path);
+    assert(pathFragments[0] == '/');
```

Comment:
It would be nice if these asserts had at least a short message associated with them, though I don't think they would be triggered that often (or ever)